src/common/utilities.ts

```diff
@@ -71,22 +71,76 @@ export async function getProjectRoot(): Promise<WorkspaceFolder> {
 }
 
 export async function installDependenciesIfNeeded(context: vscode.ExtensionContext): Promise<void> {
-    const alreadyInstalled = context.globalState.get('dependenciesInstalled', false);
+    // Install necessary dependencies for the flowcharts and telemetry
+    const alreadyInstalled = context.globalState.get(DEPENDENCIES_INSTALLED, false);
 
     if (!alreadyInstalled) {
-        const pathToScript = 'bundled/tool/install_dependencies.py';
+        const vizPathToScript = 'bundled/tool/install_viz_dependencies.py';
+        const telemetryPathToScript = 'bundled/tool/install_telemetry_dependencies.py';
         try {
-            const stdout = await callPythonScript(pathToScript, EXTENSION_ROOT_DIR, context);
-
+            const stdoutViz = await callPythonScript(vizPathToScript, EXTENSION_ROOT_DIR, context);
+            const stdoutTelemetry = await callPythonScript(telemetryPathToScript, EXTENSION_ROOT_DIR, context);
             // Check if the script output contains the success message
-            if (stdout.includes('Successfully installed')) {
-                context.globalState.update('dependenciesInstalled', true);
-                traceLog(`Python dependencies installed!`);
-                console.log('Python dependencies installed!');
+            if (stdoutViz.includes('Successfully installed')) {
+                traceLog(`Kedro-viz dependencies installed!`);
+                console.log('Kedro-viz dependencies installed!');
+            }
+            if (stdoutTelemetry.includes('Successfully installed')) {
+                traceLog(`kedro-telemetry dependencies installed!`);
+                console.log('kedro-telemetry dependencies installed!');
             }
+            context.globalState.update(DEPENDENCIES_INSTALLED, true);
```

Comment:
I think we need to set `DEPENDENCIES_INSTALLED` flag as `true` inside `stdoutViz.includes('Successfully installed')` and `stdoutTelemetry.includes('Successfully installed')`